platforms/jvm/language-groovy/src/main/java/org/gradle/api/internal/tasks/compile/ApiGroovyCompiler.java

```diff
@@ -242,7 +242,7 @@ public void compile(List<String> files, CompilationUnit cu) {
                         if (shouldProcessAnnotations) {
                             // In order for the Groovy stubs to have annotation processors invoked against them, they must be compiled as source.
                             // Classes compiled as a result of being on the -sourcepath do not have the annotation processor run against them
-                            spec.setSourceFiles(Iterables.concat(spec.getSourceFiles(), projectLayout.files(stubDir).getAsFileTree()));
+                            spec.setSourceFiles(Iterables.concat(spec.getSourceFiles(), objectFactory.fileTree().from(stubDir)));
```

Comment:
🎉 This was the only reason why workers had the `ProjectLayout` service available. An unnecessary reason. Thanks for fixing this.